sys/dev/isp/isp.c

```diff
@@ -452,103 +596,160 @@ isp_reset(ispsoftc_t *isp, int do_load_defaults)
 	isp->isp_fwrev[1] = mbs.param[2];
 	isp->isp_fwrev[2] = mbs.param[3];
 	isp->isp_fwattr = mbs.param[6];
-	isp->isp_fwattr |= ((uint64_t) mbs.param[15]) << 16;
-	if (isp->isp_fwattr & ISP2400_FW_ATTR_EXTNDED) {
-		isp->isp_fwattr |=
-		    (((uint64_t) mbs.param[16]) << 32) |
-		    (((uint64_t) mbs.param[17]) << 48);
+	if (isp->isp_fwattr & ISP_FW_ATTR_EXTNDED) {
+		isp->isp_fwattr_h = mbs.param[15];
```

Comment:
Why did you move this under ISP_FW_ATTR_EXTNDED?  According to my specs the flag covers only registers 15 and 16.

Reply:
You're right. This should be changed to
```
isp->isp_fwattr = mbs.param[6];
isp->isp_fwattr_h = mbs.param[15];
if (isp->isp_fwattr & ISP_FW_ATTR_EXTNDED) {
        isp->isp_fwattr_ext[0] = mbs.param[16];
        isp->isp_fwattr_ext[1] = mbs.param[17];
}
```
The Linux driver isn't using this flag at all. They decide on the FWI2 (firmware interface 2 ???) capability of the adapter, which is true for all 24xx+.

Reply:
fixed